python-sdk/kiota-gen.py

```diff
@@ -10,7 +10,13 @@
 import xml.etree.ElementTree as ET
 
 KIOTA_OS_NAMES = {"Windows": "win", "Darwin": "osx", "Linux": "linux"}
-KIOTA_ARCH_NAMES = {"32bit": "x86", "64bit": "x64"}
+KIOTA_ARCH_NAMES = {32: "x86", 64: "x64"}
```

Comment:
What about having:

```suggestion
KIOTA_ARCH_NAMES = {
    "x64": ["x86_64", "amd64"],
    "x86": ["x86", "i386"],
    "arm64": ["aarch64", "arm64"],
}


def get_kiota_name(machine: str) -> str | None:
    for kiota_arch, possible_names in KIOTA_ARCH_NAMES:
        if machine in possible_names:
            return kiota_arch
    else:
        return None
```

Reply:
I turned things around a bit, please have a look, I'd like to keep the desired outcome in the "value" column of the dictionary as it's probably easier/more natural to reason about.

Reply:
I was thinking about using a `NamedTuple` exactly to avoid this confusion on the usage, so something like

```py
class Arch(NamedTuple):
    kiota_name: str
    machine_names: list[str]
```

wdyt?

Reply:
now, instead of having a dict, you'd have just a list, which would make more sense as I suggested simply iterating through it (and thus, there'd be no advantage to using a dict)

Reply:
it makes sense, but I'd favor keeping things as simple as possible in this case(as opposed to as safe as possible), this Dictionary will not grow much more than its current incarnation.